homeassistant/components/media_player/aquostv.py

```diff
@@ -123,6 +132,15 @@ def set_state(self, state):
         """Set TV state."""
         self._state = state
 
+    def test_connection(self):
+        try:
+            sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
```

Comment:
All device interfacing must be handled by the pypi library. Please extract this to the library if needed.